alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -141,13 +143,38 @@ const TeamManager& RunnableObject::getTeamManager() const
     return *_teamManager;
 }
 
+void TraceRunnableObject::traceException(const std::string& exceptionOriginMethod, const std::string& details)
+{
+    std::string msg = "Exception thrown from: " + getName() + "'s " + exceptionOriginMethod + " method";
+    if (!details.empty()) {
+        msg.append(", details: " + std::move(details));
+    }
+    Logging::logFatal(LOGNAME) << msg;
+}
+
+void RunnableObject::handleException(const std::string& exceptionOriginMethod, std::exception_ptr eptr)
+{
+    std::string details;
+    try {
+        std::rethrow_exception(eptr);
+    } catch (const std::exception& e) {
+        details = e.what();
```

Comment:
We cant just print this from eptc?

Reply:
apparently not. We need to rethrow, catch & then handle the exception again